### PR TITLE
Add handling for buildModuleUrl on workers

### LINKED ISSUE #6711: GroundPolylinePrimitive._initializeTerrainHeightsWorker fails when app is served via a proxied domain
GroundPolylinePrimitive._initializeTerrainHeightsWorker tries to load in the `approximateTerrainHeights.json` from a relative path. 
> **GroundPolylinePrimitive.js**
```
ApproximateTerrainHeights.initialize('../Assets/approximateTerrainHeights.json')
```

This then calls the `buildModuleUrl` function with that path before passing it to `Resource.fetchJson`

>**ApproximateTerrainHeights.js**
```
ApproximateTerrainHeights._initPromise = Resource.fetchJson(buildModuleUrl(url)).then(function(json) {
            ApproximateTerrainHeights._terrainHeights = json;
        });
```

`buildModuleUrl` first does a check to see if it's running in node by checking `typeof document === 'undefined'` and if true, it just returns `moduleID`. 

When inside a web worker, document is undefined and the relative path is returned as it would be in node. This is fine when the code is being accessed on the same domain as it's being served on. 

But when you proxy the app, the relative path becomes `http://proxied-url.com/../Assets/approximateTerrainHeights.json` which doesn't exist as the JSON is being served on a different domain and so the `Resource.fetchJson` blows up.

### PR DESCRIPTION
**Pull request by likangning93**
Opening a preliminary PR showing that this works in non-built Cesium.
Not working in built Cesium though, and I could use a little help figuring out what's going on.

See discussion in #6716 and #6715.

Fixes #6711

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@likangning93, thanks for the pull request! Maintainers, we have a signed CLA from @likangning93, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: Source/Core/buildModuleUrl.js [hunk 1]
```diff
@@ -87,12 +90,15 @@ define([
             }
         }
 
+        var url = implementation(moduleID);
```
Comment (likangning93) [inline Source/Core/buildModuleUrl.js]: @mramato @shunter what I'm seeing is that `implementation` in built Cesium is `buildModuleUrlFromBaseUrl` on the main process but `buildModuleUrlFromRequireToUrl` on the worker, but for the worker this returns an absolute URL for non-built and a relative one for built?

Comment (likangning93) [reply Source/Core/buildModuleUrl.js]: This is easiest to reproduce using the `Polyline` Sandcastle.

**Comment by mramato** [comment]
Thanks @likangning93, I'll take a look at this later today (or over the weekend).

File: Source/Core/buildModuleUrl.js (hunk 1 above)
Comment (likangning93) [reply Source/Core/buildModuleUrl.js]: One last observation, this line does something different between built/unbuilt:
https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/TaskProcessor.js#L151

And then this https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Workers/cesiumWorkerBootstrapper.js#L227 gets set differently.

**Comment by mramato** [comment]
I pushed some changes that take a different approach.  I don't think we need to pass a "isWorker" bool of any sort, my goal is to have everything "just work" but having cesiumWorkerBootstrapper get the absolutely URL of Cesium passed in from the main thread and configure buildModuleUrl to use that instead.

I still have more work to do here, but I think it should be pretty clean.  @shunter any thoughts here? (Current code is hack, just asking about the approach).

File: Source/Scene/GroundPolylinePrimitive.js [hunk 2]
```diff
@@ -382,7 +382,7 @@ define([
             return initPromise;
         }
 
-        GroundPolylinePrimitive._initPromise = ApproximateTerrainHeights.initialize('../Assets/approximateTerrainHeights.json')
+        GroundPolylinePrimitive._initPromise = ApproximateTerrainHeights.initialize()
```
Comment (likangning93) [inline Source/Scene/GroundPolylinePrimitive.js]: Looks like we won't need `_initializeTerrainHeightsWorker` anymore then. This is only getting called from `createGroundPolylineGeometry.js`.

**Comment by hpinkos** [comment]
bump
reminder that this needs to be merged by friday

**Comment by mramato** [comment]
@likangning93 I think I have this working in all cases.  Can you put it through its paces to make sure the problem is really fixed.

As a bonus, this now works in Node.js as well.

@shunter can you take a look at the diff?

**Comment by mramato** [comment]
Whoops, looks like the built tests are failing.  I'll check it out.

**Comment by likangning93** [comment]
@mramato seems to work like a charm!

Tested on Windows with Firefox, Chrome, and Edge and on macOS and iOS with Safari.

**Comment by mramato** [comment]
OK, this should be ready.  @shunter can you do a review?  It's mostly what we talked about yesterday.

We need this in the next release.

**Comment by shunter** [comment]
Looks fine to me from the diffs.  I didn't test anything.

**Comment by mramato** [comment]
Thanks @shunter!

@likangning93 merge if you're happy.

**Comment by likangning93** [comment]
Thanks @mramato!

**Pull request merged**